### DISCUSSION TITLE
Global Unused Analyzing / Colors & Fonts fields mismatches vs Rider

### DISCUSSION THREAD

**Discussion by xCISACx** [consulo/consulo#432] [General] [answered]
Rider greys out method names for methods that haven't been used.

![Screenshot_40](https://user-images.githubusercontent.com/37281623/99133859-830f8a80-2613-11eb-9023-ec196921b35c.png)

Consulo on the left, Rider on the right.

I'd like to know if it would be possible for the same behaviour to be achieved with Consulo seeing as it is a very useful visual indicator to know if a method has been called anywhere or not.

I've also been trying to match my old Rider theme to the new Consulo one but the names and locations of the different options to change don't seem to be the same in Consulo.
Would it be possible to match the names Consulo displays so they are easier to find and match?

In Consulo, the name for changing this specific colour is Identifier, Constant does not change it.
![consulo identifier](https://user-images.githubusercontent.com/37281623/99134452-bd7a2700-2615-11eb-961c-37a46e1c8d4d.png)

In Rider, however, it's under Constant in the menu.
![rider constant](https://user-images.githubusercontent.com/37281623/99134478-d1258d80-2615-11eb-89c1-95d1eda7ec5d.png)

I'm not sure which one has the right naming, but it could help with configuring themes if the names matched, especially if Rider themes could be directly imported to Consulo.
I managed to import mine before by changing the version in the file by opening it with a text editor, but the mismatched names make it so that certain settings do not get passed on.

**Comment by VISTALL** [comment, marked as answer]
Hello.  It's global unused analyzing. It's my implementation gap in C# plugin (with control flow).  That not easy for implement - and i don't have free time for it :(

**Reply by xCISACx** [reply]
I see, I did not know it wasn't something easy to implement, that's unfortunate.

Would it still be possible to match Rider's colour configuration names?
I think Consulo is grouping different types into a single type and that's why the colours and names don't match.

I also noticed that writing something like Color.red underlines the "red" in red in Rider, but in Consulo is does not happen.
Is that difficult to implement as well?

**Reply by VISTALL** [reply]
About Color Schemes. You changed, Constant for all languages. As i saw there missed implementation for C# (under Class Fieds & Properties & Events). **Static field or property** closer - but it for all. I will check now

> I also noticed that writing something like Color.red underlines the "red" in red in Rider, but in Consulo is does not happen.
Is that difficult to implement as well?

It's specific implementation to Rider. Consulo mark **color** calls at left side (gutter).

But looks like you have buggy line-number panel. Can you say what Font and FontSize you use?

**Reply by xCISACx** [reply]
I use the Consolas font at size 15 for the Editor, I believe.
I think it's the default one for everything else since I don't have the "Use custom font" option selected under General > Appearance.

Apparently if you tick the "use custom font" box and change the size to say, 20, if you untick the box, the font changes back but the size stays the same.

**Reply by VISTALL** [reply]
About gutter, reproduced, and created issue https://github.com/consulo/consulo/issues/433

**Reply by xCISACx** [reply]
Thank you for your hard work!
Did you check the font one?

**Reply by VISTALL** [reply]
There problem of big font size. Looks like icon scale wrong, and it broking each icon in progression

**Reply by xCISACx** [reply]
Yeah, I also noticed it broke the icons somewhat.
I had to tick the box again, change the size back to 12 and hit Apply for the size to go back to what it was when the box wasn't ticked.
Can you reproduce it?

**Reply by VISTALL** [reply]
Will test.

Added option for C#. Now constant and static fields have different options

**Reply by xCISACx** [reply]
Thank you very much!
I'll keep fiddling around with the settings and let you know if I find any more that should be separate :)

**Comment by xCISACx** [comment]
Found another one:

![image](https://user-images.githubusercontent.com/37281623/99150169-990c6200-268a-11eb-9d7c-91fe4832a298.png)

Rider has an extra Property field while Consulo (I think) inherits the Property's color from static field.
They should be separate, correct?

**Reply by VISTALL** [reply]
I'm not sure. There a different settings for static field | instance field for default settings, global property I'm not sure need.

But for C# there options for set Field&Property color, but not for each of them. I'm not sure - need it or not, depends to requests

**Reply by xCISACx** [reply]
I think they should be different since seeing a property have the same colour as a static field can be quite confusing.

Changing the one from the C# section changes some properties, but not all, like the Color.red ones.
![image](https://user-images.githubusercontent.com/37281623/99150555-539d6400-268d-11eb-8918-5674c77d2305.png)

**Reply by VISTALL** [reply]
It's static :)

**Reply by xCISACx** [reply]
But even if the property itself is static, shouldn't the colour for properties be different from the static fields, even if they are static?
It just looks weird to me to see it in the same colour 😅

**Reply by VISTALL** [reply]
Fixed bug with icon panel. I will call rebuild after some time - my jenkins slow, it will rebuild after few hours.

I pushed split Field&Property highlight. Update available. After min 30, i will start full rebuild of plugins

**Reply by xCISACx** [reply]
Thank you!
I'll check it out.

**Reply by xCISACx** [reply]
The gutter bug is now fixed and my colours show up properly.
I also managed to change the property colour, thanks a lot :D

The icons are fixed as well, but the bug where you need to tick the custom font box and pick the default text size in order for the text size to be changed back instead of just unticking it and hitting Apply still stands.

**Reply by VISTALL** [reply]
Nice
